### PR TITLE
validate method for validating the specific message type

### PR DESCRIPTION
**Pull request by tulinkry**
- oop design for validating custom types
- basic test coverage
- filling the documentation with the new config type

### CODE REVIEW

**Comment by tulinkry** [comment]
And fixes #1403

File: src/org/opensolaris/opengrok/configuration/messages/StatsMessage.java [hunk 1]
```diff
@@ -43,4 +46,18 @@
         }
         return Util.statisticToJson(env.getStatistics()).toJSONString().getBytes();
     }
+
+    @Override
+    public void validate() throws Exception {
+        if (getText() == null) {
+            throw new Exception("The message must contain a text.");
+        }
+        List<String> allowed = Arrays.asList(new String[]{"get", "reload", "clean"});
```
Comment (vladak) [inline src/org/opensolaris/opengrok/configuration/messages/StatsMessage.java]: these should probably be defined only once in the file

File: src/org/opensolaris/opengrok/configuration/messages/Message.java [hunk 2]
```diff
@@ -84,6 +83,19 @@
     public abstract byte[] apply(RuntimeEnvironment env) throws Exception;
 
     /**
+     * Validate the current message and throw an exception if the message is not
+     * valid. Further implementation can override this method to get the message
+     * into a state they would expect.
+     *
+     * @throws Exception
+     */
+    public void validate() throws Exception {
```
Comment (vladak) [inline src/org/opensolaris/opengrok/configuration/messages/Message.java]: what about `className` ?

**Comment by vladak** [comment]
What about performing validation in `apply()` ?

**Comment by tulinkry** [comment]
> What about performing validation in apply() ?

good idea. fixed

File: src/org/opensolaris/opengrok/configuration/messages/StatsMessage.java (hunk 1 above)
Comment (tulinkry) [inline src/org/opensolaris/opengrok/configuration/messages/StatsMessage.java]: fixed

File: src/org/opensolaris/opengrok/configuration/messages/Message.java (hunk 2 above)
Comment (tulinkry) [inline src/org/opensolaris/opengrok/configuration/messages/Message.java]: Class name has only meaning for normal message so I don't validate it. That's a bad original design for now.

**Comment by vladak** [comment]
Looks good, squash and merge.

**Pull request merged**